dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaClusterCPUChartTask.java

```diff
@@ -0,0 +1,153 @@
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.ClouderaManagerHandle.ClouderaClusterDTO;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.hc.core5.net.URIBuilder;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The task collects hosts from Cloudera Manager <a
+ * href="https://cldr2-aw-dl-gateway.cldr2-cd.svye-dcxb.a5.cloudera.site/static/apidocs/resource_TimeSeriesResource.html">TimeSeries
+ * API</a> The chart is for whole cluster is available on {@code /cmf/home/} and {@code
+ * cmf/clusters/{clusterId}/status} pages in Cloudera Manager UI. <b/> The query to chart is written
+ * on <a
+ * href="https://docs.cloudera.com/documentation/enterprise/latest/topics/cm_dg_tsquery.html">tsquery</a>
+ * language.
+ */
+public class ClouderaClusterCPUChartTask extends AbstractClouderaManagerTask {
+  enum TimeSeriesAggregation {
+    RAW,
+    TEN_MINUTELY,
+    HOURLY,
+    SIX_HOURLY,
+    DAILY,
+    WEEKLY,
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(ClouderaCMFHostsTask.class);
+  /*
+    SELECT cpu_percent_across_hosts WHERE entityName = "1546336862" AND category = CLUSTER
+  */
+  private static final String TS_CPU_QUERY_TEMPLATE =
+      "SELECT cpu_percent_across_hosts WHERE entityName = \"%s\" AND category = CLUSTER";
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  private final int includedLastDays;
+  private final TimeSeriesAggregation tsAggregation;
+
+  public ClouderaClusterCPUChartTask() {
+    this(1, TimeSeriesAggregation.RAW);
+  }
+
+  public ClouderaClusterCPUChartTask(int includedLastDays, TimeSeriesAggregation tsAggregation) {
+    super(
+        String.format(
+            "cmf-cluster-cpu-%s-%s.jsonl",
+            includedLastDays, tsAggregation.toString().toLowerCase()));
+    this.includedLastDays = includedLastDays;
+    this.tsAggregation = tsAggregation;
+  }
+
+  @Override
+  protected Void doRun(
+      TaskRunContext context, @Nonnull ByteSink sink, @Nonnull ClouderaManagerHandle handle)
+      throws Exception {
+    CloseableHttpClient httpClient = handle.getHttpClient();
+    List<ClouderaClusterDTO> clusters = getClustersFromHandle(handle);
+
+    final String timeSeriesAPIUrl = buildTimeSeriesUrl(handle.getApiURI().toString());
+    try (Writer writer = sink.asCharSink(StandardCharsets.UTF_8).openBufferedStream()) {
+      for (ClouderaClusterDTO cluster : clusters) {
```

Comment:
This is additional data for clusters, so I think we should not fail if you can't get it for a single cluster. We should try all of them.

Reply:
Done. Removed todo comment.